lib/load-config.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+const path = require('path');
+const fs = require('fs');
+const pkgConf = require('pkg-conf');
+
+const findFileUp = pattern => (searchDir = process.cwd()) => {
+	const files = fs.readdirSync(searchDir);
+	const foundFile = files ?
+		files.find(entry => pattern instanceof RegExp ? entry.match(pattern) : entry === pattern) :
+		false;
+	if (foundFile) {
+		return path.resolve(searchDir, foundFile);
+	}
+	if (searchDir === '/') {
+		return false;
+	}
+	return findFileUp(pattern)(path.resolve(searchDir, '..'));
+};
+
+const findPackageDotJson = findFileUp('package.json');
+
+const findConfigFile = findFileUp(/ava\.config(\.\w+)?\.js/);
+
+const findConfigFiles = () => {
+	const packageFile = findPackageDotJson();
+	const configFile = findConfigFile(
+		packageFile ? path.dirname(packageFile) : process.cwd(),
```

Comment:
Let's simplify this. We can keep the current `pkg-conf` logic to find the `package.json`, if any. Afterwards, the `ava.config.js` file *must* be in the `projectDir`. We won't walk the file system to find it. (This means the `findFileUp` logic above is not necessary.)